### PR TITLE
convenience method for getting value from Subscribable

### PR DESCRIPTION
**Pull request by paulnpdev**
## What changed?
added convenience method intended for users of dynamicconfig.TypedSubscribable[T] (though possibly useful to others)
## Why?
allows access to the current value without setting up and tearing down a subscription

## How did you test it?
Added unit test

## Potential risks
Can't imagine this causing problem ... if user is currently passing nil callback it would NPE

## Documentation
Added documentation to code; shouldn't affect product docs.

## Is hotfix candidate?
No

### CODE REVIEW

File: common/dynamicconfig/collection.go [hunk 1]
```diff
@@ -308,16 +308,24 @@ func subscribe[T any](
...
 	defer c.subscriptionLock.Unlock()
 
+	// get one value immediately (note that subscriptionLock is held here so we can't race with
+	// an update)
+	init := matchAndConvert(c, key, def, cdef, convert, prec)
+	
+	// As a convenience (and for efficiency), you can pass in a nil callback; we just return the
+	// current value and skip the subscription.  The cancellation func returned is a no-op and
+	// can be called or ignored.
+	if reflect.ValueOf(callback).IsNil() {
```
Comment (dnr) [inline common/dynamicconfig/collection.go:318]: Why not `callback == nil`? It's a concrete type here, not an interface type

**Review by dnr — COMMENTED** [review]
Discussed in private, but for the pr:

I don't think we need this. You can just use Get instead of Subscribe if you want one value.

File: common/dynamicconfig/collection.go (hunk 1 above)
Comment (paulnpdev) [reply common/dynamicconfig/collection.go:318]: a func argument is by definition an interface type

Comment (dnr) [reply common/dynamicconfig/collection.go:318]: As far as I know this isn't true, there's nothing here that suggests a function has anything to do with interfaces: https://go.dev/ref/spec#Function_types

To put it concretely, how could I construct a call such that `callback == nil` is false but `reflect.ValueOf(callback).IsNil()` is true?

File: common/dynamicconfig/collection.go [hunk 2]
```diff
@@ -308,16 +308,24 @@ func subscribe[T any](
 	c.subscriptionLock.Lock()
 	defer c.subscriptionLock.Unlock()
 
+	// get one value immediately (note that subscriptionLock is held here so we can't race with
+	// an update)
+	init := matchAndConvert(c, key, def, cdef, convert, prec)
+
+	// As a convenience (and for efficiency), you can pass in a nil callback; we just return the
+	// current value and skip the subscription.  The cancellation func returned is a no-op and
+	// can be called or ignored.
```
Comment (dnr) [inline common/dynamicconfig/collection.go:317]:
```suggestion
	// current value and skip the subscription.  The cancellation func returned is nil.
```

**Review by dnr — APPROVED** [review]

**Pull request merged**